pwiz_tools/Skyline/FileUI/PeptideSearch/TransitionSettingsControl.cs

```diff
@@ -349,11 +352,30 @@ public TransitionSettings GetTransitionSettings(Form parent)
                 settings.Instrument.MaxInclusions, settings.Instrument.MinTime, settings.Instrument.MaxTime);
             Helpers.AssignIfEquals(ref instrument, settings.Instrument);
 
-            TransitionLibraryPick pick = settings.Libraries.Pick != TransitionLibraryPick.none ? settings.Libraries.Pick : TransitionLibraryPick.all;
+            var pick = settings.Libraries.Pick != TransitionLibraryPick.none && DocumentSettings.PeptideSettings.Libraries.LibrarySpecs.Any(lib => !lib.IsDocumentLibrary)
+                ? settings.Libraries.Pick
+                : TransitionLibraryPick.filter;
+            if (CountFinder != null && pick == TransitionLibraryPick.filter)
+                pick = TransitionLibraryPick.all;
             var libraries = new TransitionLibraries(ionMatchTolerance, minIonCount, ionCount, pick);
             Helpers.AssignIfEquals(ref libraries, settings.Libraries);
 
             return new TransitionSettings(settings.Prediction, filter, libraries, settings.Integration, instrument, settings.FullScan, settings.IonMobilityFiltering);
         }
+
+        private IEndCountFragmentFinder CountFinder => TransitionFilter.GetEndFragmentFinder(

```

Comment:
This needs a comment on exactly what it is doing and how it is used.  You may want a less terse expression that allows you to better comment the choices for when the values currently in the form are used and when the value from the current document is used.